pyspecdata/core.py

```diff
@@ -7536,6 +7537,10 @@ def _init_datanode(self, datanode, **kwargs):
                         + explain_error(e)
                     )
                 recordarrayofaxis = datadict["axes"][axisname]["data"]
+                if "axis_coords_units" in datadict["axes"][axisname].keys():
+                    myaxis_units[axisnumber] = datadict["axes"][axisname].__getitem__("axis_coords_units")
```

Comment:
This is not going to work in the general case. Unless you initialize this to a list of the correct length... I would start by initializing to a list of Nones (multiply a list of length 1 with element None by whatever length you want for the list)

Reply:
it is initialized above